### PR TITLE
CSHARP-799 Add category for integration tests with real clusters and optimize the test execution on appveyor

### PR DESCRIPTION
**Pull request by joao-r-reis**

### CODE REVIEW

File: src/Cassandra.IntegrationTests/Core/ClientTimeoutTests.cs [hunk 1]
```diff
@@ -20,34 +20,46 @@ public ClientTimeoutTests()
...
             Diagnostics.CassandraStackTraceIncluded = true;    
         }
-
+        
         [Test]
         public void Should_Move_To_Next_Host_For_Simple_Queries()
         {
-            var testCluster = TestClusterManager.GetNonShareableTestCluster(2, 1, true, false);
-            var socketOptions = new SocketOptions().SetReadTimeoutMillis(3000);
+            var testCluster = SimulacronCluster.CreateNew(2);
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Core/ClientTimeoutTests.cs]: Nice time saves in this fixture!

File: src/Cassandra.IntegrationTests/Ado/AdoBasicTests.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 namespace Cassandra.IntegrationTests.Data
 {
-    [TestFixture, Category("short")]
+    [TestFixture, Category("short"), Category("realcluster")]
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Ado/AdoBasicTests.cs]: I really like the idea to have a category where tests can share a cluster.

We could use a category name that carries more meaning than "realcluster"? Some suggestions:
- minimal
- basic
- singlecluster

File: src/Cassandra.IntegrationTests/SharedSimulacronClusterTest.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+namespace Cassandra.IntegrationTests
+{
+    /// <summary>
+    /// Represents a test fixture that on setup, it creates a test cluster available for all tests.
+    /// <para>
+    /// With a shared session and cluster.
+    /// </para>
+    /// </summary>
+    [TestFixture]
+    public class SharedSimulacronClusterTest
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/SharedSimulacronClusterTest.cs]: I think its a leftover.

File: src/Cassandra.IntegrationTests/Ado/AdoBasicTests.cs (hunk 2 above)
Comment (jorgebay) [reply src/Cassandra.IntegrationTests/Ado/AdoBasicTests.cs]: Its hard to name categories that serve every purpose and `realcluster` allows us to mark those that require C* clusters.
I was thinking about `minimal`, as a subgroup of `short` that would be for the 5 min runs in appveyor and also allows us to smoke test new versions of the server, but there isn't much overlap (simulacron doesn't test the server 😃 )

Nevermind, sorry for the misunderstanding.

File: src/Cassandra.IntegrationTests/SharedSimulacronClusterTest.cs (hunk 3 above)
Comment (joao-r-reis) [reply src/Cassandra.IntegrationTests/SharedSimulacronClusterTest.cs]: removed

File: src/Cassandra.IntegrationTests/Ado/AdoBasicTests.cs (hunk 2 above)
Comment (joao-r-reis) [reply src/Cassandra.IntegrationTests/Ado/AdoBasicTests.cs]: Sounds good!

**Pull request merged**